students/sikorski_marcin/Lesson_2_-_Python_basics.py

```diff
@@ -0,0 +1,130 @@
+#Lesson_3
+#Leap Year
+year = int(input("Enter year to check: "))
+
+if (year % 4 == 0) and (year % 100 != 0) or (year % 400 == 0):
+    print('LEAP')
+else:
+    print('COMMON')
+
+#Chocolate bar
+first_value_of_portion_ortion = int(input("Enter first value: "))
+second_value_of_portion = int(input("Enter second value: "))
+number_of_squares = int(input("Enter number of squares: "))
+
+if (number_of_squares %first_value_of_portion_ortion == 0 and number_of_squares /first_value_of_portion_ortion < second_value_of_portion) or (number_of_squares %second_value_of_portion== 0 and number_of_squares /second_value_of_portion < first_value_of_portion_ortion):
+    print("YES - we can divide choclate!")
+else:
+    print("NO - we can't divide chocolate")
+
+#Knight move
+first_column = int(input("Enter first column: "))
+first_row = int(input("Enter first row: "))
+second_column = int(input("Enter second column: "))
+second_row = int(input("Enter second row: "))
+
+row = abs(first_row-second_row)
+column = abs(first_column -second_column)
+
+if row == 2 and column == 1 or row == 2 and column == 1:
+    print("YES - we can move there!")
+else:
+    print("NO - we can't move there!")
+
+#Queen move
+first_column = int(input("Enter first column: "))
+first_row = int(input("Enter first row: "))
+second_column = int(input("Enter second column: "))
+second_row = int(input("Enter second row: "))
+
+row = abs(first_row-second_row)
+column = abs(first_column -second_column)
+
+
+if row == column or first_row == second_row or first_column == second_column:
+    print("YES - we can move the queen there!")
+else:
+    print("NO - we can't move queen there")
+
+#Bishop Moves
+first_column = int(input("Enter first column: "))
+first_row = int(input("Enter first row: "))
+second_column = int(input("Enter second column: "))
+second_row = int(input("Enter second row: "))
+
+row = abs(first_row -second_row)
+column = abs(first_column-second_column)
+
+if row == column:
+    print("YES - we can move bishop there!")
+else:
+    print("NO - we can't move bishop there!")
+
+#Lesson 4
+#Lost cards
+number_of_cards = int(input("Enter number of cards: "))
+sum_of_number_of_cards = (((1 + number_of_cards)/2) * number_of_cards)
+
+sum = 0
+while number_of_cards != 1:
```

Comment:
Tutaj wskazana jest zmiana pętli `while` na `for`, ponieważ jest to niezbyt eleganckie rozwiązanie.